doc/git/providers.mkd

```diff
@@ -50,6 +50,24 @@ git:
   username: 'git'
 ```
 
+#### Supported transports with Rugged
+
+For ssh support, you need to have libssh2 installed (along with the relevant dev package/headers) before you install the Rugged gem.
+
+For https support on Linux, you need to have OpenSSL installed (along with the relevant dev package/headers) before you install the Rugged gem. OS X and Windows support should automatically include https support.
+
+You can check which features your Rugged supports by using the following in irb:
+
+```ruby
+irb(main):001:0> require('rugged')
+=> true
+irb(main):002:0> Rugged.features
+=> [:threads, :https, :ssh]
```

Comment:
I would like for us to document that r10k 2.0.0 will probe for those features, but in r10k 1.5.1 users can use irb to make that check themselves.